### ISSUE TITLE
Descriptions for dividing subscription should be generated after related localizations was synced

### ISSUE THREAD

**Issue by abstractmj** [clusternet/clusternet#688]
Sometimes when the base arrives at the hub before the localization, the deployment in one member cluster may first expands from 0 to N, and then immediately shrinks to M;
0 < M < N

**Comment by dixudx** [comment]
https://github.com/clusternet/clusternet/blob/433027290481ba0777646ea337809d7e394bed39/pkg/controllermanager/deployer/deployer.go#L389-L409

Normally `Localization` objects are created and updated after `Base` objects. Then controller `Description` will begin the rendering of descriptions objects.

> Sometimes when the base arrives at the hub before the localization, the deployment in one member cluster may first expands from 0 to N, and then immediately shrinks to M;
0 < M < N

For newly created objects, there should only be from 0 to M.

But if a subscription is updated, then there will be an interim transition from N to M, just like you said. In this case, it seems there is no way to avoid such transitions, unless we introduce a new flag to indicate the outdated localizations and hold the rendering of description objects.

**Comment by abstractmj** [comment]
Normally Localization objects are created and updated after Base objects. Then controller Description will begin the rendering of descriptions objects.
----------
But we cannot guarentee the creation order between Localization and Description, so description may be already applied in member cluster before Localization is handled by hub

**Comment by dixudx** [comment]
> But we cannot guarentee the creation order between Localization and Description, so description may be already applied in member cluster before Localization is handled by hub

@abstractmj That's true for the updates of Subscriptions and Bases. That's why I think we may introduce a new flag to indicate the outdated localizations and hold the rendering of description objects.
